spec/iso8601/time_interval_spec.rb

```diff
@@ -0,0 +1,375 @@
+      expect(ISO8601::TimeInterval.new(end_time, duration).end_time).to eq((end_time + 60 * 60))
+      # This calculation is Base + Duration
+      expect(ISO8601::TimeInterval.new(duration_with_base).end_time).to eq((end_time + duration.to_f))
+    end
+  end
+
+  describe "#original_end_time" do
+    it "should return an instance of original pattern/object" do
+      duration = ISO8601::Duration.new('PT1H')
+      datetime = ISO8601::DateTime.new('2010-05-09T10:30:00Z')
+      duration_with_base = ISO8601::Duration.new('PT1H', datetime)
+
+      expect(ISO8601::TimeInterval.new(duration, datetime).original_end_time).to be_an_instance_of(ISO8601::DateTime)
+      expect(ISO8601::TimeInterval.new(datetime, duration).original_end_time).to be_an_instance_of(ISO8601::Duration)
+      expect(ISO8601::TimeInterval.new(duration_with_base).original_end_time).to be_an_instance_of(ISO8601::Duration)
+    end
+  end
+
+  describe "#to_s" do
+    it "should return the pattern if TimeInterval is initialized with a pattern" do
+      pattern = 'P1Y1M1DT0,5H/2014-05-31T16:26:10,5Z'
+      pattern2 = '2007-03-01T13:00:00Z/P1Y0,5M'
+
+      expect(ISO8601::TimeInterval.new(pattern).to_s).to eq(pattern)
+      expect(ISO8601::TimeInterval.new(pattern2).to_s).to eq(pattern2)
+    end
+
+    it "should build the pattern and return if TimeInterval is initialized with objects" do
+      duration = ISO8601::Duration.new('P1Y1M1DT0.5H')
+      datetime = ISO8601::DateTime.new('2010-05-09T10:30:12+00:00')
+      datetime2 = ISO8601::DateTime.new('2010-05-15T10:30:12+00:00')
+      duration_with_base = ISO8601::Duration.new('P1Y1M1DT0.5H', datetime)
+
+      expect(ISO8601::TimeInterval.new(duration, datetime).to_s).to eq('P1Y1M1DT0.5H/2010-05-09T10:30:12+00:00')
+      expect(ISO8601::TimeInterval.new(datetime, duration).to_s).to eq('2010-05-09T10:30:12+00:00/P1Y1M1DT0.5H')
+      expect(ISO8601::TimeInterval.new(datetime, datetime2).to_s).to eq('2010-05-09T10:30:12+00:00/2010-05-15T10:30:12+00:00')
+      expect(ISO8601::TimeInterval.new(duration_with_base).to_s).to eq('2010-05-09T10:30:12+00:00/P1Y1M1DT0.5H')
+    end
+  end
+
+  describe "compare Time Intervals" do
+    before(:each) do
+      @small = ISO8601::TimeInterval.new('2007-03-01T13:00:00Z/PT1H')
+      @big = ISO8601::TimeInterval.new('2007-03-01T13:00:00Z/PT2H')
+    end
+
+    it "should raise TypeError when compared object is not a ISO8601::TimeInterval or Numeric" do
+      expect { @small < 'Hello!' }.to raise_error(ISO8601::Errors::TypeError)
+      expect { @small > 'Hello!' }.to raise_error(ISO8601::Errors::TypeError)
+      expect { @small == 'Hello!' }.to raise_error(ISO8601::Errors::TypeError)
+    end
+
+    it "should check what interval is bigger" do
+      expect(@small > @big).to be_falsy
+      expect(@big > @small).to be_truthy
+      expect(@small > @small).to be_falsy
+      expect(@small > @small.to_f).to be_falsy
```

Comment:
I'm not really comfortable with comparing two different kind of things.  I know `Atom` currently accepts that but I may remove this by `1.0.0` to be strict with types.

I'm making some changes in your comparable implementation, I will push that soon.